0004-ecs2ext-protocol/README.md

```diff
@@ -0,0 +1,251 @@
+---
+name: Communication protocol between Endpoint Communication Service and extension service
+shortname: 5/ECS2EXT
+status: draft
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The ECS2EXT protocol is a [Kaa event-based protocol](/) extension.
+
+It is intended to solve the problem of communication between ECS and any extension service.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Message delivery confirmation._
+
+   Solutions:
+   - Protocol contains field named "status" that is used to send error message or confirm successful delivery. 
+   
+2. _Ability to determine which Avro schema should be used on extension to deserialize payload._ 
+
+   Solutions:
+   - MQTT path (e.g. "config/pull", "log/send" etc) will be passed as a `path` field in ECS2EXT protocol.
+
+3. _Ability to send payload in to directions._ 
```

Comment:
Could you elaborate on this? I don't quite understand why a single message should have payloads for both directions (simultaneously?).